rmw_cyclonedds_cpp/src/fallthrough_macro.hpp

```diff
@@ -15,15 +15,18 @@
 #ifndef FALLTHROUGH_MACRO_HPP_
 #define FALLTHROUGH_MACRO_HPP_
 
-#if __has_cpp_attribute(fallthrough) || (__cplusplus >= 201603L)
+#if __has_cpp_attribute(fallthrough) && (__cplusplus >= 201703L)
```

Comment:
The `&&` is the real fix here. I think `201603L` and `201703L` are functionally the same for this. 2016 is when it was added to the working draft, 2017 is when it was released. Whichever you prefer!

https://en.cppreference.com/w/cpp/feature_test
https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros

Reply:
I might even prefer removing the `__has_cpp_attribute(fallthrough)` and just leaving cpp version detection.

Reply:
Is the change you requested here to remove the `__has_cpp_attribute(fallthrough)` then?

Reply:
Yes, I think that would be the best move to remove the `__has_cpp_attribute(fallthrough)`, since even if it exists, we're pretending it doesn't. Also pick a `__cplusplus` version. It was not a typo, but whichever version feels better to you.

Reply:
done in 2f42980